restructure `About` section

thesuperzapper opened the pull request:
This PR resolves https://github.com/kubeflow/website/issues/2807

__It implements the following changes to the `about` section:__
- `Introduction to Kubeflow`
    - __MOVE__ to `Getting Started / Introduction to Kubeflow`
    - this page will be more visible under `Getting Started`
- `Use Cases`
    - __REMOVE__
    - this page reads like marketing material, (and is very out of date)
- `Contributing to Kubeflow`
    - __LEAVE__ 
    - can be updated in future, but is fine for now
- `Community`
    - __SPLIT UP__
    - this page fails to highlight the most important links to the user, so we should break it up into:
        - `Kubeflow Blog`: a page which describes (and links to) the [offical Kubeflow blog](https://blog.kubeflow.org/)
        - `Kubeflow Slack Workspace`: a page that describes (and provides an invite to) the official Kubeflow Slack
        - `Kubeflow Mailing List`: a page which describes (and links to) the [kubeflow-dicsuss](https://groups.google.com/g/kubeflow-discuss) mailing list
        - `Kubeflow Community Call`: a page that describes the weekly community meeting, and how to join it
        - `Kubeflow Working Groups`: a page that lists and describes the Kubeflow Working Groups
            - links to the [kubeflow/community](https://github.com/kubeflow/community/tree/master/wg-automl) pages 
            - links to somewhere which aggregates all meeting calendars
- `Google Summer of Code`
    - __REMOVE__
    - this event has finished, and it's confusing to keep this page
-  `Docs`
    - __REMOVE__
    - this page does not provide useful information
- `Style Guide for the Kubeflow Docs`
    - __RENAME__  to `Documentation Style Guide`
    - this is still a very useful page (but possibly we should highlight it somewhere)
- `Events Calendar`
    - __REMOVE__
    - this page has not been updated in over a year

---

__Leaving the following structure for `about`:__
```
About:
- Kubeflow Blog
- Kubeflow Slack Workspace
- Kubeflow Mailing List
- Kubeflow Community Call
- Kubeflow Working Groups
- Contributing to Kubeflow
- Documentation Style Guide
```

__Adding the following page to `getting started`:__
```
Getting Started:
- Installing Kubeflow
```

---

__It also does the following:__
- updates all old slack & mailing-list links to the new `About` pages
- cleans up some of the text on the pages not removed from `About`

kramachandran commented:
@Bobgy, @zijianjoy  -- This PR contains some documentation clean up that I am interested in, particularly the removal of references to GSoC.  Could you review and approve. 

I also think the movement for the Introducing Kubeflow page from this section to the Getting Started section makes a ton of sense.

Bobgy commented:
Hi @thesuperzapper, thank you for this restructure!
All the changes look great to me, except that I don't quite understand the rationale of splitting up the community page.
It seems that now "blog", "slack", "community meeting" etc pages typically only have a few sentences. Why not grouping them together in a community page?

These info are also super useful for contributors, so what about increasing the community page's visibility by adding links from the "contributing to Kubeflow" page. https://www.kubeflow.org/docs/about/contributing/ Contributing page is also one of the top entrances in kubeflow.org.

zijianjoy commented on config.toml:
```diff
@@ -214,35 +214,24 @@ no = 'Sorry to hear that. Please <a href="https://github.com/kubeflow/website/is
 	icon = "fa fa-envelope"
-        desc = "Discussion and help from your fellow users"
-[[params.links.user]]
-	name = "Calendar"
```
I believe this is still useful as in https://github.com/kubeflow/community/tree/master/calendar. I join Kubeflow community meeting every week using this calendar. Why do we remove this?

thesuperzapper replied:
I have readded it (but made it a link to `/docs/about/community-call/`).

I think that's better than linking directly to it, so we don have to keep two links up to date if/when we change calanders.

zijianjoy replied:
Sounds good, it makes sense!

zijianjoy commented on config.toml:
```diff
@@ -214,35 +214,24 @@ no = 'Sorry to hear that. Please <a href="https://github.com/kubeflow/website/is
 [[params.links.user]]
 	name = "Mailing list"
-	url = "https://groups.google.com/forum/#!forum/kubeflow-discuss"
+	url = "/docs/about/kubeflow-mailing-list"
```
I guess the hyper link here is one of the reasons for breaking up sections in `Community` page to different pages.

But I would suggest to use hashtag instead: `/docs/about/community/#mailing-lists` in here, which makes it easier for people to see all community contact approach if they navigate from this footer icon.

thesuperzapper replied:
I have changed it back to a single community page.

zijianjoy commented on content/en/_index.html:
```diff
@@ -173,12 +171,12 @@ <h5 class="card-title text-white section-head">Community</h5>
     <h4 class="section-head">Join our community</h4>
     <p>
-      Want to contribute to Kubeflow? <a href="/docs/about/community/">Join our community</a>! Check out upcoming meetings, get involved in discussions on our mailing list or on Slack, and suggest exciting new integrations.
+      Check out the <a href="/docs/about/kubeflow-community-call/">weekly community call</a>, get involved in discussions on the <a href="/docs/about/kubeflow-mailing-list/">mailing list</a> or chat with others on the <a href="/docs/about/kubeflow-slack-workspace/">Slack Workspace</a>!
```
As you might have already noticed, we have many pages in kubeflow.org, it will be awesome to make relevant information more grouped, than having to navigate among many pages like this sentence tries to do.

thesuperzapper replied:
See my thoughts here: https://github.com/kubeflow/website/pull/3007#issuecomment-943762457

thesuperzapper replied:
I have changed it back to a single community page.

zijianjoy commented on content/en/docs/about/_index.md:
```diff
@@ -1,5 +1,5 @@
 +++
 title = "About"
-description = "Information about Kubeflow software, community, docs, and events."
+description = "About Kubeflow and its community"
```
After this PR's refactoring, I believe this doc area has become `About Kubeflow community` than `About Kubeflow and its community`. The Kubeflow introduction has been moved to `Getting Started` doc area.

thesuperzapper replied:
I would still say its "information about Kubeflow", (e.g. the docs style guide, and contributing)

zijianjoy replied:
I see, we can discuss about the description separately. Let's get the doc restructuring in this PR merged in first.

thesuperzapper commented on config.toml:
```diff
@@ -214,35 +214,24 @@ no = 'Sorry to hear that. Please <a href="https://github.com/kubeflow/website/is
-	icon = "fa fa-calendar"
-        desc = "Discussion and help from your fellow users"
-
+    desc = "Join the Kubeflow Mailing List!"
```
Turns out this file is indented with tabs, I have fixed that line now.

zijianjoy commented:
Thank you @Bobgy for the review!

annajung commented:
Not a reviewer, but just putting my thought out here as well since this was discussed during the community meeting today. I think a simple table of contents on top of the community page could also really help. & considering that slack, mailing list, a community meeting is already in the footer, I prefer a single page as well to avoid multiple clicks. :)

thesuperzapper commented:
@Bobgy @james-jwu  it looks like people have voted to combine the pages, I will refactor the PR to recombine the community pages.

thesuperzapper commented:
@Bobgy @james-jwu I have updated the PR with the required changes, it should be ready to merge now.

---

The refactored `About` section.

![Screenshot 1](https://user-images.githubusercontent.com/5735406/143018440-3630dcec-e9a4-4252-835d-1ee5c9d1b2d3.png)

---

The new `About / Community` page.

![Screenshot 2](https://user-images.githubusercontent.com/5735406/143018467-2cdd2bdc-451a-41fa-93b2-e48d61560699.png)

---

The updated `Getting Started` page.

![Screenshot 3](https://user-images.githubusercontent.com/5735406/143018585-652607aa-321d-4563-b8a8-e02a9199687a.png)

james-jwu commented:
Looks good to me, but I'll let @zijianjoy take a final look.

Bobgy commented:
Hi @thesuperzapper, @zijianjoy is on vacation for Thanksgiving, let's wait for him to come back.
Can you resolve merge conflicts first?
/lgtm

thesuperzapper commented:
@Bobgy @zijianjoy I have rebased to fix merge conflicts.

Also, I have renamed the `Getting Started / Kubeflow Overview` page to `Getting Started / Architecture`, for clarity because the `Getting Started / Introduction` page now exists.

zijianjoy commented on content/en/docs/distributions/minikf/minikf-vagrant.md:
```diff
@@ -26,8 +26,7 @@ Join the discussion on the
 
-To join the Kubeflow Slack workspace, please [request an
-invite](https://kubeflow.slack.com/join/shared_invite/zt-cpr020z4-PfcAue_2nw67~iIDy7maAQ).
+To join the Kubeflow Slack workspace, please see [here](/docs/about/community/#slack).
```
NIT: Usually we don't use `this`/`here`/etc. when adding hyperlink. So the sentence here can be:

Refer to [Community Slack Guide](/docs/about/community/#slack) for joining the Kubeflow Slack workspace.

thesuperzapper replied:
Done in https://github.com/kubeflow/website/pull/3007/commits/028e09ad53948f87a1f93675ccc9ae553aba8dc6

zijianjoy reviewed (commented):
/lgtm

Thank you so much for organizing the `About` section! I only have one NIT comment.

thesuperzapper commented:
@zijianjoy I have fixed the nit you had, this should be ready to merge now!

zijianjoy commented:
/approve

Thank you for the cleanup and restructuring the `About` section!

thesuperzapper commented:
@zijianjoy, @james-jwu or @Bobgy this still needs an `/lgtm` (even though its had an `/approve`)

zijianjoy commented:
/lgtm
